src/main/java/dev/jbang/source/sources/KotlinSource.java

```diff
@@ -23,6 +23,13 @@ public KotlinSource(String script, Function<String, String> replaceProperties) {
 		super(script, replaceProperties);
 	}
 
+	@Override
+	protected List<String> collectBinaryDependencies() {
+		final List<String> allDependencies = super.collectBinaryDependencies();
+		allDependencies.add("org.jetbrains.kotlin:kotlin-stdlib:" + getKotlinVersion());
```

Comment:
not a fan of these but if needed lets do it. got example of script that fails without it ? then should add as test.

Reply:
@evanchooly would like to add this - just looking for examples to try to trigger error without this in place?

Reply:
I'll get that today.  Just been distracted...

Reply:
https://gist.github.com/evanchooly/9d271ab06b86d775bb0aef19b21b7a41

Reply:
ugh.  that "description" in the gist is not easy to ready.  Here it is:

Attached is the script I used to test the various combinations as well as the script that was causing problems.  Hopefully, this will be enough to repro the issue but if not, i'm happy to fill in any gaps.  Drop both files in the root folder in my PR branch and run `rebuild.sh` and you should see what I'm talking about.

Reply:
in the script, it runs the test script via the PATH and an explicit reference to the executable because at one point I had noticed a difference in behavior where one approach would successfully run the script but the other would fail.  I haven't seen it since so it might look like extra noise but I was trying to suss out that difference.

Reply:
when i run the sample .kt file I get info about missing main?

with and without your changes.

expected ? 

```
❯ jbang --fresh DriverSnapshot.kt
[jbang] Downloading Kotlin 2.0.0. Be patient, this can take several minutes...
[jbang] Installing Kotlin 2.0.0...
[jbang] Resolving dependencies...
[jbang]    org.jetbrains.kotlin:kotlin-stdlib:2.0.0
[jbang] Dependencies resolved
[jbang] Building jar for DriverSnapshot.kt...
[jbang] [ERROR] no main class deduced, specified nor found in a manifest
[jbang] Run with --verbose for more details
```

Reply:
I *don't* get that actually just running code on that PR branch of mine.  Did you rebase it?  Should I?

when I the command you use above with `--verbose`, I get this output:

```
[jbang] [0:185] jbang version 0.117.1.1
[jbang] [0:190] Resolving resource ref: DriverSnapshot.kt
[jbang] [0:192] Resolved resource ref as: DriverSnapshot.kt (cached as: /home/evanchooly/dev/jbang/DriverSnapshot.kt)
[jbang] [0:216] Building as fresh build explicitly requested.
[jbang] [0:221] Resolving artifact(s): com.github.zafarkhaja:java-semver:0.9.0, com.fasterxml.jackson.core:jackson-databind:2.15.2, com.fasterxml.jackson.dataformat:jackson-dataformat-xml:2.15.2, org.jetbrains.kotlin:kotlin-stdlib:2.0.0
[jbang] [0:223] Repositories: central=https://repo1.maven.org/maven2/
[jbang] [0:223] Resolving dependencies...
[jbang] [0:304]    com.github.zafarkhaja:java-semver:0.9.0
[jbang] [0:316]    org.sonatype.oss:oss-parent:9
[jbang] [0:326]    com.fasterxml.jackson.core:jackson-databind:2.15.2
[jbang] [0:328]    com.fasterxml.jackson:jackson-base:2.15.2
[jbang] [0:330]    com.fasterxml.jackson:jackson-bom:2.15.2
[jbang] [0:331]    com.fasterxml.jackson:jackson-parent:2.15
[jbang] [0:332]    com.fasterxml:oss-parent:50
[jbang] [0:337]    org.junit:junit-bom:5.9.2
[jbang] [0:340]    com.fasterxml.jackson.core:jackson-annotations:2.15.2
[jbang] [0:343]    com.fasterxml.jackson.core:jackson-core:2.15.2
[jbang] [0:347]    com.fasterxml.jackson.dataformat:jackson-dataformat-xml:2.15.2
[jbang] [0:352]    org.codehaus.woodstox:stax2-api:4.2.1
[jbang] [0:355]    com.fasterxml.woodstox:woodstox-core:6.5.1
[jbang] [0:358]    org.jetbrains.kotlin:kotlin-stdlib:2.0.0
[jbang] [0:359]    org.jetbrains:annotations:13.0
[jbang] [0:365] Dependencies resolved
[jbang] [0:389] Resolved artifact(s): com.github.zafarkhaja:java-semver:jar:0.9.0=/home/evanchooly/.m2/repository/com/github/zafarkhaja/java-semver/0.9.0/java-semver-0.9.0.jar, com.fasterxml.jackson.core:jackson-databind:jar:2.15.2=/home/evanchooly/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.15.2/jackson-databind-2.15.2.jar, com.fasterxml.jackson.core:jackson-annotations:jar:2.15.2=/home/evanchooly/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.15.2/jackson-annotations-2.15.2.jar, com.fasterxml.jackson.core:jackson-core:jar:2.15.2=/home/evanchooly/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.15.2/jackson-core-2.15.2.jar, com.fasterxml.jackson.dataformat:jackson-dataformat-xml:jar:2.15.2=/home/evanchooly/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-xml/2.15.2/jackson-dataformat-xml-2.15.2.jar, org.codehaus.woodstox:stax2-api:jar:4.2.1=/home/evanchooly/.m2/repository/org/codehaus/woodstox/stax2-api/4.2.1/stax2-api-4.2.1.jar, com.fasterxml.woodstox:woodstox-core:jar:6.5.1=/home/evanchooly/.m2/repository/com/fasterxml/woodstox/woodstox-core/6.5.1/woodstox-core-6.5.1.jar, org.jetbrains.kotlin:kotlin-stdlib:jar:2.0.0=/home/evanchooly/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib/2.0.0/kotlin-stdlib-2.0.0.jar, org.jetbrains:annotations:jar:13.0=/home/evanchooly/.m2/repository/org/jetbrains/annotations/13.0/annotations-13.0.jar
[jbang] [0:415] Building jar for DriverSnapshot.kt...
[jbang] [0:415] Compile: /home/evanchooly/.jbang/cache/kotlincs/2.0.0/kotlinc/bin/kotlinc -classpath /home/evanchooly/.m2/repository/com/github/zafarkhaja/java-semver/0.9.0/java-semver-0.9.0.jar:/home/evanchooly/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.15.2/jackson-databind-2.15.2.jar:/home/evanchooly/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.15.2/jackson-annotations-2.15.2.jar:/home/evanchooly/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.15.2/jackson-core-2.15.2.jar:/home/evanchooly/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-xml/2.15.2/jackson-dataformat-xml-2.15.2.jar:/home/evanchooly/.m2/repository/org/codehaus/woodstox/stax2-api/4.2.1/stax2-api-4.2.1.jar:/home/evanchooly/.m2/repository/com/fasterxml/woodstox/woodstox-core/6.5.1/woodstox-core-6.5.1.jar:/home/evanchooly/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib/2.0.0/kotlin-stdlib-2.0.0.jar:/home/evanchooly/.m2/repository/org/jetbrains/annotations/13.0/annotations-13.0.jar -d /home/evanchooly/.jbang/cache/jars/DriverSnapshot.kt.59b8e759dd020b268e5c5a257c5c863fac4d01f71ba3b6066c7a351195a740a5/classes /home/evanchooly/dev/jbang/DriverSnapshot.kt
[jbang] [2:806] Looking for JDK: 0
[jbang] [2:816] Using JDK: 11 (11.0.23+9, current, /home/evanchooly/.sdkman/candidates/java/11.0.23-tem)
[jbang] [2:817] Looking for JDK: 0
[jbang] [2:817] Using JDK: 11 (11.0.23+9, current, /home/evanchooly/.sdkman/candidates/java/11.0.23-tem)
[jbang] [2:818] Package: /home/evanchooly/.sdkman/candidates/java/11.0.23-tem/bin/jar cfme /home/evanchooly/.jbang/cache/jars/DriverSnapshot.kt.59b8e759dd020b268e5c5a257c5c863fac4d01f71ba3b6066c7a351195a740a5/DriverSnapshot.jar /tmp/jbang-manifest4102225341703617928mf DriverSnapshotKt -C /home/evanchooly/.jbang/cache/jars/DriverSnapshot.kt.59b8e759dd020b268e5c5a257c5c863fac4d01f71ba3b6066c7a351195a740a5/classes .
[jbang] [2:894] Deleting file /tmp/jbang-manifest4102225341703617928mf
[jbang] [2:895] Deleting folder /home/evanchooly/.jbang/cache/jars/DriverSnapshot.kt.59b8e759dd020b268e5c5a257c5c863fac4d01f71ba3b6066c7a351195a740a5/classes
[jbang] [2:896] Deleting folder /home/evanchooly/.jbang/cache/jars/DriverSnapshot.kt.59b8e759dd020b268e5c5a257c5c863fac4d01f71ba3b6066c7a351195a740a5/generated
[jbang] [2:897] Looking for JDK: 0
[jbang] [2:897] Using JDK: 11 (11.0.23+9, current, /home/evanchooly/.sdkman/candidates/java/11.0.23-tem)
[jbang] [2:897] Looking for JDK: 0
[jbang] [2:897] Using JDK: 11 (11.0.23+9, current, /home/evanchooly/.sdkman/candidates/java/11.0.23-tem)
[jbang] [2:898] run: /home/evanchooly/.sdkman/candidates/java/11.0.23-tem/bin/java -classpath /home/evanchooly/.jbang/cache/jars/DriverSnapshot.kt.59b8e759dd020b268e5c5a257c5c863fac4d01f71ba3b6066c7a351195a740a5/DriverSnapshot.jar:/home/evanchooly/.m2/repository/com/github/zafarkhaja/java-semver/0.9.0/java-semver-0.9.0.jar:/home/evanchooly/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.15.2/jackson-databind-2.15.2.jar:/home/evanchooly/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.15.2/jackson-annotations-2.15.2.jar:/home/evanchooly/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.15.2/jackson-core-2.15.2.jar:/home/evanchooly/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-xml/2.15.2/jackson-dataformat-xml-2.15.2.jar:/home/evanchooly/.m2/repository/org/codehaus/woodstox/stax2-api/4.2.1/stax2-api-4.2.1.jar:/home/evanchooly/.m2/repository/com/fasterxml/woodstox/woodstox-core/6.5.1/woodstox-core-6.5.1.jar:/home/evanchooly/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib/2.0.0/kotlin-stdlib-2.0.0.jar:/home/evanchooly/.m2/repository/org/jetbrains/annotations/13.0/annotations-13.0.jar DriverSnapshotKt
5.2.0-SNAPSHOT
```

Reply:
doh! error was between keyboard and monitor  - was trying to run an empty file (doh!) - now it fails/works as you outline.